src/averageAges.js

```diff
@@ -15,11 +15,20 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  let ages = men.map(person => person.died - person.born);
+
+  const centuryPersons = men.filter(person =>
+    Math.ceil(person.died / 100) === century);
+
+  if (century) {
+    ages = centuryPersons.map(person => person.died - person.born);
+  }
```

Comment:
you can avoid map iteration, if statement... 
```suggestion
  const men = people
    .filter(({ sex, died }) => century
      ? Math.ceil(died / 100) === century && sex === 'm' : sex === 'm');
```